Hw spring tests

Iskamele opened the pull request.

Iskamele committed:
Added spring-context and hsqldb.

Iskamele committed:
Changed access modifier.

Iskamele committed:
Added tests cases for dao.impl package.

Iskamele committed:
Added tests cases for security package.

Iskamele committed:
Added tests cases for service.impl package.

Iskamele committed:
Added tests cases for validation package.

Iskamele committed:
Added data-stub.

Rommelua commented on src/test/java/mate/academy/security/CustomUserDetailsServiceTest.java:
```diff
@@ -0,0 +1,59 @@
+    @Test
+    void loadUserByUsername_usernameNotFound_NotOk() {
+        //arrange
+        UsernameNotFoundException exception =
```
Move the UsernameNotFoundException assertion to the act & assert section, as it's part of the testing behavior.

Rommelua commented on src/test/java/mate/academy/security/CustomUserDetailsServiceTest.java:
```diff
@@ -0,0 +1,59 @@
+
+    @Test
+    void loadUserByUsername_usernameNotFound_NotOk() {
+        //arrange
```
Add a Mockito.when statement to mock the UserService.findByEmail method for the NOT_EXISTING_EMAIL to return an empty Optional.

Rommelua commented on src/test/java/mate/academy/dao/impl/AbstractTest.java:
```diff
@@ -0,0 +1,139 @@
+
+    protected String[] packages() {
+        return null;
+    }
```
Consider returning an empty array instead of null for better null-safety.

Rommelua commented on src/test/java/mate/academy/dao/impl/UserDaoImplTest.java:
```diff
@@ -0,0 +1,126 @@
+    @Test
+    void findByEmail_notExistingEmail_NotOk() {
+        //assert
+        Assertions.assertThrows(Exception.class,
```
Instead of asserting on exception, use `assertFalse(userDao.findByEmail(NOT_EXISTING_EMAIL).isPresent())` to check if the Optional is empty.

Rommelua commented on src/test/java/mate/academy/dao/impl/UserDaoImplTest.java:
```diff
@@ -0,0 +1,126 @@
+
+        //assert
+        Assertions.assertEquals(EMAIL, actual.get().getEmail());
+        Assertions.assertEquals(PASSWORD, actual.get().getPassword());
```
Use `assertTrue(actual.isPresent())` before calling `actual.get()` to ensure the Optional is not empty.

Rommelua commented on src/test/java/mate/academy/dao/impl/UserDaoImplTest.java:
```diff
@@ -0,0 +1,126 @@
+    @Test
+    void findById_notExistingId_NotOk() {
+        //assert
+        Assertions.assertThrows(Exception.class,
```
Instead of asserting on exception, use `assertFalse(userDao.findById(NOT_EXISTING_ID).isPresent())` to check if the Optional is empty.

Rommelua commented on src/test/java/mate/academy/dao/impl/UserDaoImplTest.java:
```diff
@@ -0,0 +1,126 @@
+    @Test
+    void findByEmail_nullEmail_NotOk() {
+        //assert
+        Assertions.assertThrows(Exception.class,
```
Instead of asserting on exception, use `assertFalse(userDao.findByEmail(null).isPresent())` to check if the Optional is empty.

Rommelua commented on src/test/java/mate/academy/dao/impl/UserDaoImplTest.java:
```diff
@@ -0,0 +1,126 @@
+        Optional<User> actual = userDao.findById(expected);
+
+        //assert
+        Assertions.assertEquals(expected, actual.get().getId());
```
Use `assertTrue(actual.isPresent())` before calling `actual.get()` to ensure the Optional is not empty.

Rommelua commented on src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java:
```diff
@@ -0,0 +1,69 @@
+                () -> roleDao.getRoleByName(null).orElseThrow());
+    }
+
+    private void assertRoleEquals(Role excepted, Role actual) {
```
Why do you need it? Role must have overridden equels() method

Iskamele replied:
I didn't want to change the source code unnecessarily. My task was to write the tests.

Rommelua commented on src/test/java/mate/academy/security/jwt/JwtTokenProviderTest.java:
```diff
@@ -0,0 +1,116 @@
+        userDetailsService = Mockito.mock(UserDetailsService.class);
+        jwtTokenProvider = new JwtTokenProvider(userDetailsService);
+
+        Field fieldSK = JwtTokenProvider.class.getDeclaredField("secretKey");
+        fieldSK.setAccessible(true);
+        fieldSK.set(jwtTokenProvider, SECRET_KEY);
+        Field fieldViM = JwtTokenProvider.class.getDeclaredField("validityInMilliseconds");
+        fieldViM.setAccessible(true);
+        fieldViM.setLong(jwtTokenProvider, MILLISECONDS);
```
use Spring reflection utilities to set  private fields in tests

Rommelua commented on src/test/java/mate/academy/security/jwt/JwtTokenProviderTest.java:
```diff
@@ -0,0 +1,116 @@
+    private final List<String> roleList = new ArrayList<>();
+    private JwtTokenProvider jwtTokenProvider;
+    private UserDetailsService userDetailsService;
+    private String token;
```
I do not think you need it as class field
```suggestion
```

Iskamele commented on src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java:
```diff
@@ -0,0 +1,69 @@
+    void getRoleByName_nullNameOk() {
+        //assert
+        Assertions.assertThrows(DataProcessingException.class,
+                () -> roleDao.getRoleByName(null).orElseThrow());
```
More likely because I tested RoleDao based on the ready-made UserDao, where the tests without orElseThrow did not pass. My fault.
It's not very obvious.

Iskamele committed:
Overridden equals() method.

Iskamele committed:
packages() will return an empty array instead of null.

Iskamele committed:
Exception.class has been replaced by DataProcessingException.class.
Removed assertRoleEquals() method.
Removed unnecessary orElseThrow() methods.

Iskamele committed:
Replaced all Exception.class DataProcessingException.class.
Removed unnecessary orElseThrow().
AssertThrows() replaced by assertFalse() where needed.

Iskamele committed:
Removed String token variable, now it is declared locally in methods.

Iskamele committed:
Fixed loadUserByUsername_usernameNotFound_NotOk() method.

Iskamele committed:
Fixed getRoleByName_roleByNotExistName_NotOk and getRoleByName_roleByNullName_NotOk methods. They now check the class under test.
